packages/remirror__extension-text-color/src/text-color-extension.ts

```diff
@@ -115,6 +115,18 @@ export class TextColorExtension extends MarkExtension<TextColorOptions> {
     return this.store.commands.removeMark.original({ type: this.type, ...options, expand: true });
   }
 
+  /**
+   * Get the color for the selection
+   */
+  @helper()
+  getTextColorForSelection(selection?: Selection): Helper<string> {
+    if (!selection) return this.options.defaultColor;
```

Comment:
I suspect we need to check we have a text selection here?

```ts
import { isTextSelection } from '@remirror/core';
```

```suggestion
    if (!isTextSelection(selection) || selection.empty) {
      return this.options.defaultColor;
    }
```